cmd/main.go

```diff
@@ -163,12 +169,13 @@ func main() {
 	}
 
 	lProv := &models.DefaultLocalProvider{
-		ProviderBaseURL:        DefaultProviderURL,
-		MapPreferencePersister: preferencePersister,
-		ResultPersister:        resultPersister,
-		SmiResultPersister:     smiResultPersister,
-		TestProfilesPersister:  testConfigPersister,
-		GenericPersister:       dbHandler,
+		ProviderBaseURL:              DefaultProviderURL,
+		MapPreferencePersister:       preferencePersister,
+		ResultPersister:              resultPersister,
+		SmiResultPersister:           smiResultPersister,
+		TestProfilesPersister:        testConfigPersister,
+		PerformanceProfilesPersister: performanceProfilePersister,
```

Comment:
Educate me: why is performanceProfilePersister separate from resultPersister or testPersister? Does each persister represent a file on disk?

Reply:
@leecalcote, performanceProfilePersister only concerns itself with persisting the performance profiles while the resultPersister concerns itself with persisting only the performance results. However I am not too sure about the testConfigPersister which was already present in the meshery code base.

And yes, each persister gets its own file on the user's file system.

Reply:
What factor most greatly influences the division between persisters? 

- logical grouping, which might mean that these two logically-related constructs should be persisted in the same data file. ￼

- Timing of when data needs to be written to disk. 


Considering that a test result only exists in context of a profile, should these two persisters be combined?￼

// @kumarabd

Reply:
@leecalcote profiles and results are still on different persister as all though results are associated with the profiles and it's existence is now only valid in context of a performance profile, they are two different entities from the implementation point of view.

There are still APIs that perform action on results in bulk (like date range queries or getting total results count) which benefit from keeping these two entities separate internally. The files in someway represent two different SQL DB tables, one which stores all of the results while holding a reference to the performance profile it belongs to.

Another reason to keep the persister is to let the older APIs work as intended as they rely on this persister. These older APIs may still be getting used in mesheryctl which is another meshery server client.

Reply:
Thanks for this explanation @utkarsh-pro